pkg/registration/status.go

```diff
@@ -18,16 +18,22 @@ const (
 )
 
 type statusRequest struct {
-	Hostname string            `json:"hostname"`
-	Info     SystemInformation `json:"hwinfo"`
+	Hostname string `json:"hostname"`
+
+	requestWithInformation
 }
 
 // Returns the registration status for the system pointed by the authorized
 // connection.
-func Status(conn connection.Connection, hostname string, info SystemInformation) (StatusCode, error) {
+func Status(conn connection.Connection, hostname string, systemInformation SystemInformation, extraData ExtraData) (StatusCode, error) {
```

Comment:
I moved to our own definition for `SystemInformation` and `ExtraData`. Both are type alias to the `map[string]any` construct.

Btw. my linter complains about writing `map[string]interface{}` vs `map[string]any`, so I switched to it